campaign-launcher/interfaceV2/src/providers/Web3AuthProvider.tsx

```diff
@@ -0,0 +1,142 @@
+import {
+  createContext,
+  FC,
+  PropsWithChildren,
+  useContext,
+  useState,
+  useEffect,
+} from 'react';
+
+import { useAccount, useSignMessage } from 'wagmi';
+
+import { REFRESH_FAILURE_EVENT } from '../api/httpClient';
+import recordingApi from '../api/recordingApi';
+import { tokenManager, TokenData } from '../utils/TokenManager';
+
+type Nonce = 'signup' | string;
+
+type Web3AuthContextType = {
+  isAuthenticated: boolean;
+  isLoading: boolean;
+  signIn: () => Promise<void>;
+  logout: () => Promise<void>;
+};
+
+const Web3AuthContext = createContext<Web3AuthContextType>(
+  {} as Web3AuthContextType
+);
+
+const getNonce = async (address: `0x${string}` | undefined) => {
+  const response = await recordingApi.post<Nonce>(`/auth/nonce`, { address });
+  return response.data; 
+};
+
+export const Web3AuthProvider: FC<PropsWithChildren> = ({ children }) => {
+  const [isAuthenticated, setIsAuthenticated] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+  const { signMessageAsync } = useSignMessage();
+  const { isConnected, address } = useAccount();
+
+  const signIn = async () => {
+    setIsLoading(true);
+    try {
+      const nonce = await getNonce(address);
+      const signature = await signMessageAsync({
+        message: JSON.stringify(nonce),
+      });
+      const authResponse = await recordingApi.post<TokenData>('/auth', {
+        address,
+        signature,
+      });
+
+      tokenManager.setTokens({
+        access_token: authResponse.data.access_token,
+        refresh_token: authResponse.data.refresh_token,
+      });
+      setIsAuthenticated(true);
+    } catch(e) {
+      setIsAuthenticated(false);
+      console.error(e);
```

Comment:
Atm it's going to be full axios error object (with config, that also includes auth header with token) and it can be hijacked from the console logs, so we should avoid writing full error, but only some meaningful things useful for debug purpose